napari/layers/image/experimental/octree.py

```diff
@@ -0,0 +1,197 @@
+"""Octree class.
+"""
+import numpy as np
+from scipy import ndimage as ndi
+
+# from enum import IntEnum
+# class ChildIndex(IntEnum):
+#    Upper
+
+
+def _create_tiles(array: np.ndarray, tile_size: int) -> np.ndarray:
+    """
+    Return an NxM array of (tile_size, tile_size) ndarrays except the edge
+    tiles might be smaller if the array did not divide evenly.
+    """
+    if array.ndim != 3:
+        raise ValueError(f"Unexpected array dimension {array.ndim}")
+    (rows, cols, _) = array.shape
+
+    tiles = []
+
+    row = 0
+    while row < rows:
```

Comment:
A lot of the tooling for dividing arrays into chunks exists in dask -- I'm pretty sure the thing you are doing here is done in n dimensions without while loops in the `slices_from_chunks` function in `dask.array.core` https://github.com/dask/dask/blob/45fd9092049a61c1d43067a0132574b3fe715153/dask/array/core.py#L206

Reply:
Yeah. We haven't really had the discussion of using dask inside the renderer itself, as in everything turns into dask whether it started as dask or not. Ask of today I'm assuming we don't want to do that. But that's a valid conversation to have, even soon-ish.

Reply:
That function I referenced doesn't use dask arrays at all. It's just a convenience function to generate slices for dividing an array into chunks.

Reply:
This specific code of building a tree from a single image is just for testing. Almost all multi-scale viewing will not go through this code, because it will be a pre-chunked format.

Reply:
That's all fine, I'm just letting you know that the _specific thing you are doing in this loop_ (slice an array into chunks) is already implemented via the dask core utils, so you could remove some of your code and just use / copy the dask function, if you want. For myself, whenever I'm breaking an array into little chunks I often find clean functions for that in the dask array core utils, even when I'm not working with dask arrays.

Reply:
Yes that's good to know about, thanks.